proto/redpanda/api/dataplane/v1alpha3/ai_agent.proto

```diff
@@ -0,0 +1,684 @@
+    (buf.validate.field).string.max_len = 16384,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "System instructions that define how the AI agent should behave and respond."}
+  ];
+
+  // LLM configuration settings for the AI agent.
+  Provider provider = 5 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  int32 dimensions = 6 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.field_behavior) = IMMUTABLE,
+    (buf.validate.field).required = true
+  ];
+  string model = 7 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.field_behavior) = IMMUTABLE,
+    (buf.validate.field).required = true
+  ];
+
+  // List of Redpanda MCP server IDs that this AI agent can use.
+  repeated string mcp_server_ids = 8;
+
+  //  // Map of MCP servers that this AI agent can use.
+  //  map<string, MCPServer> mcp_servers = 8 [
+  //    (buf.validate.field).map.keys.string = {
+  //      pattern: "^[A-Za-z0-9_-]+$"
+  //      max_len: 64
+  //    },
+  //    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Map of MCP server configurations that this AI agent can use for tool execution."}
+  //  ];
+
+  // Tags are key-value pairs that can be assigned to an AI agent resource.
+  // They help organize AI agents and enable filtering when listing them.
+  map<string, string> tags = 9 [(buf.validate.field).map = {
+    max_pairs: 16
+    values: {
+      string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+    }
+  }];
+
+  // The current AI agent state.
+  State state = 10 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // URL to connect to the AI agent endpoint.
+  string url = 11 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Response configuration.
+  oneof response_configuration {
+    // Text response configuration.
+    TextResponseConfig text_response = 12;
+
+    // JSON response configuration.
+    JsonResponseConfig json_response = 13;
+  }
+
+  message Provider {
+    message OpenAI {
+      string api_key = 1 [
+        (google.api.field_behavior) = REQUIRED,
+        (buf.validate.field).required = true,
+        (buf.validate.field).string.pattern = "^\\$\\{secrets\\.[A-Za-z0-9_]+\\}$"
+      ];
+    }
+
+    oneof provider {
+      option (buf.validate.oneof).required = true;
+      OpenAI openai = 1;
+    }
+  }
+
+  // State of the AI agent.
+  enum State {
+    STATE_UNSPECIFIED = 0;
+    // The AI agent is starting.
+    STATE_STARTING = 1;
+    // The AI agent is running and ready to handle requests.
+    STATE_RUNNING = 2;
+    // The AI agent is in the process of stopping.
+    STATE_STOPPING = 3;
+    // The AI agent is stopped and in paused state.
+    STATE_STOPPED = 4;
+    // The AI agent encountered an error.
+    STATE_ERROR = 5;
+  }
+
+  // MCP transport types.
+  enum MCPTransport {
+    MCP_TRANSPORT_UNSPECIFIED = 0;
+    MCP_TRANSPORT_STREAMABLE = 1;
+    MCP_TRANSPORT_SSE = 2;
```

Comment:
i think we can skip this one, if we support only redpanda cloud remote mcp for now, we do not need this sort of configurability